prov/sockets/src/sock_cntr.c

```diff
@@ -512,12 +513,11 @@ int sock_cntr_open(struct fid_domain *domain, struct fi_cntr_attr *attr,
 		goto err;
 
 	if (attr == NULL)
-		memcpy(&_cntr->attr, &sock_cntr_add, sizeof(sock_cntr_attr));
+		memcpy(&_cntr->attr, &sock_cntr_attr, sizeof(sock_cntr_attr));
```

Comment:
Can you separate this into its own patch?  This looks like a bug fix.

Reply:
Sure, I can.
Btw, there are multiple places where a function is declared to return `size_t`, but there are branches in the function that returns `-1`. Should I separate them as well?